backends/gdx-backend-lwjgl/src/com/badlogic/gdx/backends/lwjgl/LwjglGL20.java

```diff
@@ -36,33 +36,55 @@
 
 /** An implementation of the {@link GL20} interface based on LWJGL. Note that LWJGL shaders and OpenGL ES shaders will not be 100%
  * compatible. Some glGetXXX methods are not implemented.
- *
  * @author mzechner */
 class LwjglGL20 implements com.badlogic.gdx.graphics.GL20 {
-	private ByteBuffer buffer = null;
-	private FloatBuffer floatBuffer = null;
-	private IntBuffer intBuffer = null;
-
-	private void ensureBufferCapacity (int numBytes) {
-		if (buffer == null || buffer.capacity() < numBytes) {
-			buffer = com.badlogic.gdx.utils.BufferUtils.newByteBuffer(numBytes);
-			floatBuffer = buffer.asFloatBuffer();
-			intBuffer = buffer.asIntBuffer();
+	/** ClassX : support for thread safety
+	 * @author dar */
+	static final class NioThreadLocal extends ThreadLocal<NioThreadLocalData> {
+		/*
+		 * @see java.lang.ThreadLocal#initialValue()
+		 */
+		@Override
+		protected NioThreadLocalData initialValue () {
+			return new NioThreadLocalData();
+		}
+	}
+
+	/** ClassX : support for thread safety
+	 * @author dar */
+	static final class NioThreadLocalData {
+		public ByteBuffer buffer;
+		public FloatBuffer floatBuffer;
+		public IntBuffer intBuffer;
+	}
+
+	// nio thread local static instance
+	private final static NioThreadLocal nioThreadLocal = new NioThreadLocal();
```

Comment:
This should be `private static final` (per https://stackoverflow.com/questions/16731240/what-is-a-reasonable-order-of-java-modifiers-abstract-final-public-static-e/16732059#16732059).

Reply:
Oh my God, haha... I believe I have made this point myself at least once and I have been told that `static public final` would be the way to do it here. There is no behavior difference.

Reply:
private static final should be the order. Apart from that, there is no behavior difference.